rfcs/severity.md

```diff
@@ -0,0 +1,130 @@
+# Severity of hints
+
+## Current status
+
+Hints have an associated severity (`off`, `warning`, `error`) that can be set
+via a `.hintrc` file when using the CLI. E.g.:
+
+```json
+{
+    "hints": {
+        "hint-name": "error"
+    }
+}
+```
+
+The browser extension or the online scanner do not have the possibility to change
+the severity. What's more, a lot of users rely in the provided configurations
+(e.g.: `configuration-web-recommended`).
+
+The problem with this is that it is difficult for developers to know what is really
+important and should be fixed right away, and what is not as relevant:
+
+> If everything is important, then nothing is.
+
+On top of that:
+
+* there are hints that report several message and not all of them have
+the same importance
+* some hints do not respect the user's severity for some of the messages
+
+This RFC proposes significant changes to the way the severity is treated in webhint
+so we provide more actionable feedback and help developers be more succesfull.
+
+## Proposal
+
+**TL;DR;**
+
+Have hints report with the same severity accross all the different mediums (cli,
+vs code, browser extension, and online scanner) by:
+
+1. Changing the available severity levels for hints.
+1. Making hints responsible of setting the severity for each report.
+1. Letting users change only if a hint is enable or disable (plus some extra
+   configuration in the applicable hints).
+
+### Hint severity level
+
+The idea is to get closer to the severity options available in
+[VS Code](https://code.visualstudio.com/api/references/vscode-api#DiagnosticSeverity):
+
+| Severity | Description |
+| -------- | ----------- |
+| off      | Disabled, no reporting will come from it |
+| error    | Something that the user should get fixed right away |
```

Comment:
I'll add the following images to the doc so we have a better idea of how it will look like in VS Code

VS Code Error
![vs code error](https://user-images.githubusercontent.com/606594/64741804-bf46d880-d4ae-11e9-91a1-1db1b60d29a5.png)

VS Code Warning
![vs code warning](https://user-images.githubusercontent.com/606594/64741875-ff0dc000-d4ae-11e9-8668-eecea56d418d.png)

VS Code Information
![vs code information](https://user-images.githubusercontent.com/606594/64741926-2cf30480-d4af-11e9-88fe-01e4d9d7d36c.png)

VS Code Hint
![vs code hint](https://user-images.githubusercontent.com/606594/64741963-6166c080-d4af-11e9-825e-47bb2ca60a26.png)